agent/engine/host_resource_manager.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package engine contains the core logic for managing tasks
+
+package engine
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/ecs_client/model/ecs"
+	"github.com/aws/amazon-ecs-agent/agent/logger"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+)
+
+// HostResourceManager keeps account of each task in
+type HostResourceManager struct {
+	hostResource     map[string]*ecs.Resource
+	consumedResource map[string]*ecs.Resource
+
+	taskConsumed map[string]bool //task.arn to boolean whether host resources consumed or not
```

Comment:
Nit - if we are going to be using this map as a set (i.e. we only ever care about whether the key exists), we can use `map[string]struct{}` to indicate that the value doesn't matter. But current implementation is fine too.